### PR TITLE
Fix panic in FutexKey::load_val when no read access

### LINKED ISSUE #1334: Reachable unwrap in `FutexKey::load_val()`
### Describe the bug
There is a reachable unwrap in `FutexKey::load_val()` at kernel/src/process/posix_thread/futex.rs:333 when call `futex` with memory region without read access.

https://github.com/asterinas/asterinas/blob/53ce7df53ccb497395e87f8ee5261d0c3a3db98e/kernel/src/process/posix_thread/futex.rs#L333

### To Reproduce
1. Compile a program which calls `futex` with memory region without read access.: 
```C
#include <errno.h>
#include <linux/futex.h>
#include <stdint.h>
#include <stdio.h>
#include <sys/syscall.h>
#include <sys/mman.h>
#include <unistd.h>

int main() {
  void *addr1 = mmap(NULL, 4096, PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
  void *addr2 = mmap(NULL, 1048576, PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
  syscall(SYS_futex, addr1, 0x189, 95, NULL, addr2, 0xff49);
  perror("futex");

  return 0;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior

Asterinas reports unwrap panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.8.1
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 8bfbdf6

### Logs
```
panicked at /root/asterinas/kernel/src/process/posix_thread/futex.rs:333:53:
called `Result::unwrap()` on an `Err` value: Error { errno: EFAULT, msg: None }
Printing stack trace:
   1: fn 0xffffffff887e71b0 - pc 0xffffffff887e721a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa8e80;

   2: fn 0xffffffff887e6fc0 - pc 0xffffffff887e7110 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa8ee0;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa9030;

   4: fn 0xffffffff889c4810 - pc 0xffffffff889c4892 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa9040;

   5: fn 0xffffffff889a3910 - pc 0xffffffff889a3a3b / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa90d0;

   6: fn 0xffffffff8846dd30 - pc 0xffffffff8846ded4 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa91d0;

   7: fn 0xffffffff8846ba80 - pc 0xffffffff8846c3ad / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa92d0;

   8: fn 0xffffffff8846ba50 - pc 0xffffffff8846ba75 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa9870;

   9: fn 0xffffffff884e2d00 - pc 0xffffffff884e32c8 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa98a0;

  10: fn 0xffffffff885766d0 - pc 0xffffffff8859f093 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007caa9f30;

  11: fn 0xffffffff88576310 - pc 0xffffffff8857639e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007cac03d0;

  12: fn 0xffffffff884b2230 - pc 0xffffffff884b2daf / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007cac0570;

  13: fn 0xffffffff883b93e0 - pc 0xffffffff883b93ee / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007cac0f90;

  14: fn 0xffffffff88859bf0 - pc 0xffffffff88859c06 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007cac0fb0;

  15: fn 0xffffffff888349a0 - pc 0xffffffff88834a09 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007cac0fd0;


     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883d3a7; rbp                0x0; rsp 0xffff80007cac1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:113
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 73ctk84we8vz1kw7hv2sr00er:?
(  4) ??:?
(  5) ??:?
(  6) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:1102
(  7) /root/asterinas/kernel/src/process/posix_thread/futex.rs:44
(  8) /root/asterinas/kernel/src/process/posix_thread/futex.rs:23
(  9) /root/asterinas/kernel/src/syscall/futex.rs:46
( 10) /root/asterinas/kernel/src/syscall/mod.rs:220
( 11) /root/asterinas/kernel/src/syscall/mod.rs:328
( 12) /root/asterinas/kernel/src/thread/task.rs:69
( 13) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 14) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 15) /root/asterinas/ostd/src/task/mod.rs:175
make: *** [Makefile:172: run] Error 1
```

### PR DESCRIPTION
**Pull request by Marsman1996**
Fix #1334

### CODE REVIEW

**Comment by Marsman1996** [comment]
@boterinas rerun

**Comment by tatetian** [comment]
@StevenJiang1110 Could you review the changes? Thanks!

File: kernel/src/process/posix_thread/futex.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ pub fn futex_wait_bitset(
     // lock futex bucket ref here to avoid data race
     let mut futex_bucket = futex_bucket_ref.lock();
 
-    if futex_key.load_val() != futex_val {
+    if futex_key.load_val()? != futex_val {
```
Comment (StevenJiang1110) [inline kernel/src/process/posix_thread/futex.rs:44]: I use your test program from #1334, and Linux program reports `EAGAIN`. If you use `?` here, Asterinas will report `EFAULT`. Maybe you can try the following form
```rust
if ! futex_key.load_val().is_ok_and(|val| val == futex_val) {
    return_errno_with_message!(Errno::EAGAIN, "futex value cannot be accessed");
}
```

Comment (Marsman1996) [reply kernel/src/process/posix_thread/futex.rs]: Modified🫡

**Review by StevenJiang1110 — APPROVED** [review]
LGTM!

**Comment by Marsman1996** [comment]
@boterinas rerun

**Pull request merged**